akka-camel/src/main/scala/akka/camel/Producer.scala

```diff
@@ -128,7 +128,7 @@ trait ProducerSupport extends Actor with CamelSupport {
     protected def produce(endpoint: Endpoint, processor: SendProcessor, msg: Any, pattern: ExchangePattern): Unit = {
       // Need copies of sender reference here since the callback could be done
       // later by another thread.
-      val producer = self
+      val producer = context.parent
```

Comment:
This looks like the right thing to do. Thanks for good description in the issue.

One question. What could cause the `ChildProducer` actor to be slow? Is there something that is blocking there that I don't see? It calls
```
processor.process(xchg.exchange, new AsyncCallback
```

Could that be blocking the actor in some cases?

The reason I'm asking is that if it's blocking we should make sure that the `ChildProducer` is running on a dedicated dispatcher, or make it easy to configure that. Now it will run on the default dispatcher and it would be bad to block threads of that.